File: src/lib/api/adapters/web/latex.ts
// Web LaTeX Adapter - HTTP API
import { apiClient } from '../../client';
import type { LaTeXCompileRequest, LaTeXCompileResponse, LaTeXOperations } from '../../types';

export class WebLatexAdapter implements LaTeXOperations {
	async compileLatex(request: LaTeXCompileRequest): Promise<LaTeXCompileResponse> {
		try {
			const response = await apiClient.post<LaTeXCompileResponse>('/api/latex/compile', request);
			
			return response;
		} catch (error) {
			console.error('Web LaTeX compilation failed:', error);
			
			// Handle HTTP errors
			const errorMessage = error instanceof Error ? error.message : 'Unknown compilation error';
			
			return {
				success: false,
				message: 'LaTeX compilation failed',
				errors: [errorMessage]
			};
		}
	}

	async findMainLatexFile(): Promise<string> {
		try {
			const response = await apiClient.get<{ success: boolean; main_file?: string; message?: string }>('/api/latex/find-main');
			
			if (!response.success) {
				throw new Error(response.message || 'Failed to find main LaTeX file');
			}
			
			return response.main_file || '';
		} catch (error) {
			console.error('Web LaTeX find main file failed:', error);
			throw error;
		}
	}

	async setAutoCompile(enabled: boolean): Promise<void> {
		try {
			await apiClient.post('/api/latex/set-auto-compile', { enabled });
		} catch (error) {
			console.error('Failed to set auto compile:', error);
			throw error;
		}
	}

	async setMainFile(filePath: string | null): Promise<void> {
		try {
			await apiClient.post('/api/latex/set-main-file', { filePath });
		} catch (error) {
			console.error('Failed to set main file:', error);
			throw error;
		}
	}
}

File: backend/src/transport/handlers/latex.rs
use crate::model::LaTeXCompileRequest;
use crate::svc::LaTeXService;
use axum::{
    debug_handler,
    extract::State,
    http::StatusCode,
    response::Json,
    Json as RequestJson,
};
use serde_json::{json, Value};
use std::sync::Arc;
use tracing::{error, info};

#[debug_handler]
pub async fn compile_latex(
    State(service): State<Arc<LaTeXService>>,
    RequestJson(request): RequestJson<LaTeXCompileRequest>,
) -> Result<Json<Value>, StatusCode> {
    info!("LaTeX compilation request received with provider: {:?}", request.provider);
    
    match service.compile_workspace(request).await {
        Ok(response) => {
            if response.success {
                info!("LaTeX compilation completed successfully");
                Ok(Json(json!({
                    "success": true,
                    "message": response.message,
                    "output_file": response.output_file
                })))
            } else {
                error!("LaTeX compilation failed: {}", response.message);
                Ok(Json(json!({
                    "success": false,
                    "message": response.message,
                    "errors": response.errors.unwrap_or_default()
                })))
            }
        }
        Err(e) => {
            error!("LaTeX compilation service error: {}", e);
            Err(StatusCode::INTERNAL_SERVER_ERROR)
        }
    }
}

#[debug_handler]
pub async fn find_main_latex_file(
    State(service): State<Arc<LaTeXService>>,
) -> Result<Json<Value>, StatusCode> {
    info!("Finding main LaTeX file request received");
    
    match service.find_main_tex_file().await {
        Ok(tex_file_path) => {
            let relative_path = tex_file_path.strip_prefix(service.get_workspace_path())
                .unwrap_or(&tex_file_path)
                .to_string_lossy()
                .to_string();
            
            info!("Found main LaTeX file: {}", relative_path);
            Ok(Json(json!({
                "success": true,
                "main_file": relative_path
            })))
        }
        Err(e) => {
            error!("Failed to find main LaTeX file: {}", e);
            Ok(Json(json!({
                "success": false,
                "message": e.to_string()
            })))
        }
    }
}

#[debug_handler]
pub async fn latex_status(
    State(service): State<Arc<LaTeXService>>,
) -> Result<Json<Value>, StatusCode> {
    let snapshot = service.get_snapshot().await;
    Ok(Json(json!({
        "success": true,
        "latex": snapshot
    })))
}